openchain/consensus/obcpbft/pbft-core.go

```diff
@@ -840,60 +762,25 @@ func (instance *pbftCore) executeOne(idx msgID) bool {
 	if digest == "" {
 		logger.Info("Replica %d executing/committing null request for view=%d/seqNo=%d",
 			instance.id, idx.v, idx.n)
+		// Note, that it is not possible for a null request to be a multiple of the checkpoint interval, so no checkpointing checking is needed in this path
```

Comment:
As I think more about this, I actually think this is a false assertion.

When a primary is faulty, and a view change occurs.  The primary has assigned requests to sequence numbers, but has chosen to skip some sequence numbers.  At view change, the latest checkpoint quorum certificate is selected to work from, and the missing requests are filled in with nulls.

The erroneous logic was that if we had requests beyond the next checkpoint sequence number, then we would have simply chosen that checkpoint as the basis for our new view, but there is no reason to believe that the Primary ever assigned a request to that checkpoint sequence number.

I'd be open to suggestions as to how to address this.  I think the obvious, but incorrect solution is to add an API to retrieve a snapshot ID from the executor.  But this seems like the wrong solution to me.  The problem being that it introduces a dependency on the current state of execution, so it would also need to include a sequence number, and it might block for a very long time, might be missed because of state transfer, etc.

My suggestions would be that we could extend the notion of a 'Null Request' to the executor, so that it does not perform an execution but does send back a checkpoint ID.  In this way it gets into the execution queue normally, and otherwise follows the normal execution/checkpoint path.